internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -228,61 +259,182 @@ func (cmi *ClusterManagerImpl) stop() error {
+		log.Debug("ClusterManagerImpl::getActiveMVs: local cluster map not loaded")
+		return nil
+	}
+	activeMVs := make(map[string]dcache.MirroredVolume)
+	for mvName, mv := range cmi.localMap.MVMap {
+		if mv.State == dcache.StateOnline {
+			activeMVs[mvName] = mv
+		}
+	}
+	return activeMVs
 }
 
 // getCacheConfig implements ClusterManager.
 func (cmi *ClusterManagerImpl) getCacheConfig() *dcache.DCacheConfig {
-	return nil
+	if cmi.localMap == nil {
+		log.Debug("ClusterManagerImpl::getCacheConfig: local cluster map not loaded")
+		return nil
+	}
+	return &cmi.localMap.Config
 }
 
 // getDegradedMVs implements ClusterManager.
-func (c *ClusterManagerImpl) getDegradedMVs() []dcache.MirroredVolume {
-	return make([]dcache.MirroredVolume, 0)
+func (cmi *ClusterManagerImpl) getDegradedMVs() map[string]dcache.MirroredVolume {
+
+	if cmi.localMap == nil {
+		log.Debug("ClusterManagerImpl::getDegradedMVs: local cluster map not loaded")
+		return nil
+	}
+	degradedMVs := make(map[string]dcache.MirroredVolume)
+	for mvName, mv := range cmi.localMap.MVMap {
+		if mv.State != dcache.StateOnline {
+			degradedMVs[mvName] = mv
+		}
+	}
+	return degradedMVs
+}
+
+// getMyRVs implements ClusterManager
+func (cmi *ClusterManagerImpl) getMyRVs() map[string]dcache.RawVolume {
+	if cmi.localMap == nil {
+		log.Debug("ClusterManagerImpl::getMyRVs: local cluster map not loaded")
+		return nil
+	}
+
+	myRvs := make(map[string]dcache.RawVolume)
+	for name, rv := range cmi.localMap.RVMap {
+		if rv.NodeId == cmi.nodeId {
+			myRvs[name] = rv
+		}
+	}
+	return myRvs
 }
 
 // getRVs implements ClusterManager.
-func (c *ClusterManagerImpl) getRVs(mvName string) []dcache.RawVolume {
-	return make([]dcache.RawVolume, 0)
+func (cmi *ClusterManagerImpl) getRVs(mvName string) map[string]string {
+	if cmi.localMap == nil {
+		log.Debug("ClusterManagerImpl::getRVs: local cluster map not loaded")
+		return nil
+	}
+	mv, ok := cmi.localMap.MVMap[mvName]
+	if !ok {
+		log.Debug("ClusterManagerImpl::getRVs: no mirrored volume named %s", mvName)
+		return nil
+	}
+	return mv.Rvs
 }
 
-func (c *ClusterManagerImpl) isOnline(nodeId string) bool {
+func (cmi *ClusterManagerImpl) isOnline(nodeId string) bool {
+	if cmi.localMap == nil {
+		log.Debug("ClusterManagerImpl::isOnline: local cluster map not loaded")
+		return false
+	}
+	for _, rv := range cmi.localMap.RVMap {
+		if rv.NodeId == nodeId {
+			return rv.State == dcache.StateOnline
+		}
+	}
+	log.Debug("ClusterManagerImpl::isOnline: node %s not found", nodeId)
 	return false
 }
 
 // lowestNumberRV implements ClusterManager.
-func (c *ClusterManagerImpl) lowestNumberRV(rvNames []string) []string {
-	return make([]string, 0)
+func (c *ClusterManagerImpl) lowestNumberRV(rvNames []string) string {
+	log.Debug("ClusterManagerImpl::lowestNumberRV: evaluating lowest number rvName %v", rvNames)
+	lowestNumberRv := ""
+	var min int
+	for _, name := range rvNames {
+		num, _ := strconv.Atoi(strings.TrimPrefix(name, "rv"))
```

Comment:
any function that returns error and some other value, we cannot check the other value w/o making sure err is nil